src/extension.rs

```diff
@@ -19,6 +19,12 @@ pub enum CompressionFormat {
     Zip,   // .zip
 }
 
+impl CompressionFormat {
+    pub fn is_archive_format(&self) -> bool {
+        matches!(self, Tar | Tgz | Tbz | Tlzma | Tzst | Zip)
```

Comment:
I wonder if this non-exhaustive pattern matching (with the implicit `_` wildcard) is a great choice, because we can forget to add a new format to this list.

Did not questioned this in the review process because I know people consider not matching every variant very convenient.

Reply:
I agree, proper match without any `_` would be better in the long term.
Though since this has been merged I urge you to take a look at #133